pkg/destroy/openstack/openstack.go

```diff
@@ -386,7 +386,12 @@ func deleteRouters(opts *clientconfig.ClientOpts, filter Filter, logger logrus.F
 		for _, fip := range allFIPs {
 			_, err := floatingips.Update(conn, fip.ID, floatingips.UpdateOpts{}).Extract()
 			if err != nil {
-				logger.Fatalf("%v", err)
+				// Ignore the error if the resource cannot be found and exit with an appropiate message if it's another type of error
+				if _, ok := err.(gophercloud.ErrDefault404); !ok {
+					logger.Fatalf("Updating floating IP %q for Router %q failed: %v", fip.ID, router.ID, err)
+					os.Exit(1)
```

Comment:
This is an extremely oppinionated API.  I'd much rather return an `error` that the caller can wrap and bubble up or otherwise handle as they see fit.

Reply:
I agree 100%.

However, that early exit is a behaviour that I don't touch in this PR (the added `os.Exit(1)` is purely cosmetic, as `logger.Fatalf` was exiting already). In this PR, I only change the behaviour in case the error is a 404.

I will be happy to delegate error handling to the caller in another PR, once we can prioritise refactoring this whole `destroy` code.